gbl-toolfile.spec

```diff
@@ -0,0 +1,25 @@
+### RPM external gbl-toolfile 1.0
+Requires: gbl
+%prep
+
+%build
+
+%install
+
+mkdir -p %i/etc/scram.d
+cat << \EOF_TOOLFILE >%i/etc/scram.d/gbl.xml
+<tool name="gbl" version="@TOOL_VERSION@">
+  <info url="https://www.wiki.terascale.de/index.php/GeneralBrokenLines"/>
+  <lib name="GBL"/>
+  <client>
+    <environment name="GBL_BASE" default="@TOOL_ROOT@"/>
+    <environment name="LIBDIR" default="$GBL_BASE/lib"/>
+    <environment name="INCLUDE" default="$GBL_BASE/include"/>
+  </client>
+  <runtime name="ROOT_INCLUDE_PATH" value="$INCLUDE" type="path"/>
+  <use name="root_cxxdefaults"/>
```

Comment:
@smuzaffar I have added these two lines because I have seen them in all other spec files, but the meaning of them is not really clear to me.
Are they actually required, if one has no ROOT dependency?

Reply:
@ghellwig , for now its fine to have them. They are only needed if headers files of this package are used (directly or indirectly) in root dictionaries.

Reply:
Thanks for the explanation. I think they won't be needed in root dictionaries, but let's review that after I integrated this external in the CMSSW code base.
Anyway, it looks like it does not really hurt to have those lines just in case, right?

Reply:
root uses ROOT_INCLUDE_PATH to find the headers files, so more items in it can slow down the search. So if you are sure that headers of this package is not going to be part of root dict then feel free to drop these.

By the way, as you have eigen dependency so please add eigen depdency in tool file too. 
```
<use name="eigen"/>
```
This will instruct scram to also add eigen include path when someone uses gbl to build/link.

Reply:
@smuzaffar regarding the dictionaries I am not 100% sure as I am sometimes surprised by how ROOT works. Maybe you can guide me a bit?
The headers that are in GBL are currently all in `Alignment/ReferenceTrajectories/interface`.
Since there is nothing in the package related to dictionaries, I assume we can indeed drop it, but as I said, I am not 100% sure.

Anyway, I'll add the eigen dependency also to the tool file.

Reply:
ROOT_INCLUDE_PATH is needed by rootcling (for relocatability)

```
Do not store the headers' directories in the dictionary
Instead, rely on the environment variable $ROOT_INCLUDE_PATH at runtime
```

https://github.com/root-project/root/blob/94a28b7c2552f3428a9605649aa6d9cde22da056/core/dictgen/src/rootcling_impl.cxx#L86
https://github.com/root-project/root/blob/cf9ce8347d28a4eeaca2ffa01670d83e5907207d/core/clingutils/src/TClingUtils.cxx#L4837